### DISCUSSION TITLE
Potiential improvement that I don't have time to benchmark but it looks promising

### DISCUSSION THREAD

**Discussion by mikefaille** [ggml-org/llama.cpp#1512] [Ideas]
In the Makefile, I append to the CFLAGS variable `-march=native      -ffast-math` for a 5.1 bit model to run as fast as if I run the 4.1 bit model.

I needed to share this information although I didn't had time to toughly test it.

To compare my potential improvement for `-march=native`, my CPU model is : i7-8700 CPU.

Please, give a feedback if you have time to do a more complete proof of concept than mine. 

Thank you,
Michael

**Comment by hungerf3** [comment]
The makefile already uses -march=native -mtune=native or the equivalent for most platforms (-mcpu=native for arm; for example) 

As for -ffast-math; it probably isn't a good idea. Currently, it appears that memory bandwidth, rather than computation speed, is the limiting factor on most systems.  I suspect you won't see any speed-up, and you might see issues. Try building with that flag, and carrying out a long conversation, then compare the interaction with the same seed and a build without -ffast-math.
